### ISSUE TITLE
redirect_uri with keycloak 18 causes an error

### ISSUE THREAD

**Issue by d-koppenhagen** [dsb-norge/vue-keycloak-js#141]
Using `@https://github.com/dsb-norge@2.1.4-beta` with keycloak 18 will produce an error when trying to logout:
```
invalid param redirect_uri
```

The Breaking Change in Keycloak is described in the [Release Notes](https://www.keycloak.org/2022/04/keycloak-1800-released) - Section "Migration from 17.0 -> OpenID Connect Logout".

The problem can be fixed by updating `keycloak-js` to version `18.0.0`.

**Comment by baltom** [comment]
As previously mentioned in other issues we are keeping keycloak-js at the current version because of redhat-sso using said version. So it won't be updated until there is a redhat-sso version with corresponding keycloak version.

In the release notes you linked there is also a way to make `redirect_uri` work with 18+ versions of keycloak

```


There is a backwards compatibility option, which allows your application to still use the old format of the redirect_uri parameter.

You can enable this parameter when you start the server by entering the following command:

     bin/kc.[sh|bat] --spi-login-protocol-openid-connect-legacy-logout-redirect-uri=true start

With this configuration, you can still use the format with the redirect_uri parameter. Note the confirmation screen will be needed if the id_token_hint is omitted.

```

**Comment by hendisantika** [comment]
Thanks @baltom ... You help me a lot.

**Comment by joe105598055** [comment]
I meet the same problem.
In my case, I downgrade keycloak@18.0.1 to keycloak@17.0.1.
That works for me.

https://quay.io/repository/keycloak/keycloak?tab=tags&tag=17.0.1

**Comment by hendisantika** [comment]
You don't need to downgrade @joe105598055 . You need to upgrade your Keycloak-js libs. It also solved the issue.

**Comment by robson90** [comment]
@hendisantika was right. I updated Keycloak-JS and it worked.

**Comment by julioAzocar** [comment]
el logout se me arreglo y me funciono asi 
Valid redirect URIs : http://localhost:4200/*
Valid post logout : http://localhost:4200/*
Web origins :  http://localhost:4200/*

**Comment by theolevisage** [comment]
@baltom If I'm not mistaken, redhat recently upgraded to sso@7.6 which uses keycloak@18.0.0
https://access.redhat.com/articles/2342881

**Comment by baltom** [comment]
@theolevisage yeah, it is indeed. So we can bump up the keycloak-js dependency. And won't need the workaround to support old format

**Comment by JavaPantry** [comment]
Is this issue fixed? I upgrade vue-keycloak-js to latest "@dsb-norge/vue-keycloak-js": "^2.1.5-beta", on keycloak-18.0.0 still receive error log message " Parameter 'redirect_uri' no longer supported. Please use 'post_logout_redirect_uri' with 'id_token_hint' for this endpoint."

I'm not sure how to upgrade `keycloak-js to version 18.0.0` in my package.json.

Thank you for your help.

**Comment by katzekro** [comment]
> el logout se me arreglo y me funciono asi Valid redirect URIs : http://localhost:4200/* Valid post logout : http://localhost:4200/* Web origins : http://localhost:4200/*

Gracias, tambien me funciono.

**Comment by cdecinkoKnight** [comment]
I'm using keycloak-js 19.0.3 and getting the indicated error with Keycloak 21.x
